### PR TITLE
item-matrix: Support recursion via intermediates

### PR DESCRIPTION
**Pull request by JasperCraeghs**
- [x] implementation
- [x] testing
- [x] documentation

### CODE REVIEW

File: doc/integration_test_report.rst [hunk 1]
```diff
@@ -506,6 +509,24 @@ Traceability via intermediate items
     :nocaptions:
     :stats:
     :coveredintermediates:
+    :coverage: <79
+
+.. warning due to below item-matrix: coverage is not 100% due to child RQT not covered
```
Comment (Letme) [inline doc/integration_test_report.rst:514]:
```suggestion
.. warning in below item-matrix: coverage is not 100% due to child RQT not covered
```

File: doc/requirements.rst [hunk 2]
```diff
@@ -99,6 +99,10 @@ Requirements for mlx.traceability
 .. item:: RQT-DUMMY Dummy requirement that is not covered by a test
     :fulfilled_by: DESIGN-ATTRIBUTES DESIGN-ITEMIZE
 
+.. item:: RQT-DUMMY_CHILD Child of the dummy requirement that is not covered by a test
+    :depends_on: RQT-DUMMY
+.. :validated_by: ITEST-DUMMY_CHILD
```
Comment (Letme) [inline doc/requirements.rst:104]: Hm, I assume this is to highlight that there is no `validated_by` relation?

File: doc/integration_test_report.rst (hunk 1 above)
Comment (JasperCraeghs) [reply doc/integration_test_report.rst]: Applied in c9f9916d8189981636ed802fe4b9d72f07f03cab

File: doc/requirements.rst (hunk 2 above)
Comment (JasperCraeghs) [reply doc/requirements.rst]: No. I hope this change clarifies things: c9f9916d8189981636ed802fe4b9d72f07f03cab

**Review by Letme — APPROVED** [review]

**Pull request merged**